### PR TITLE
[Improvement-11857][Spark] Remove the spark version of spark task

### LINKED ISSUE #11857: [Improvement][Spark] Remove the spark version of spark task
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently, the spark version is misleading.
截屏2022-09-08 15 24 34

This is not the spark version that DS currently supports. E.g., DS can also run the spark task of `SPARK-3.x.x`.

The spark version selected by the user only determines the environment variables used and the final command executed as below:

image

And also there is a bug that `spark-sql` can only be executed by `SPARK2`, see https://github.com/apache/dolphinscheduler/pull/11721

So why not just remove the spark version, like other task
And use `{SPARK_HOME}/bin/spark_submit` and `{SPARK_HOME}/bin/spark-sql`

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by rickchengx**
## Purpose of the pull request

close: #11857 

Currently, the spark version is misleading.
截屏2022-09-08 15 24 34

This is not the spark version that DS currently supports. E.g., DS can also run the spark task of `SPARK-3.x.x`.

The spark version selected by the user only determines the environment variables used and the final command executed as below:

image

And also there is a bug that `spark-sql` can only be executed by `SPARK2`, see https://github.com/apache/dolphinscheduler/pull/11721

So why not just remove the spark version, like other task
And use `{SPARK_HOME}/bin/spark_submit` and `{SPARK_HOME}/bin/spark-sql`

## Brief change log

 use `{SPARK_HOME}/bin/spark_submit` and `{SPARK_HOME}/bin/spark-sql`

## Verify this pull request

manually tested

image

截屏2022-09-08 16 51 41

### CODE REVIEW

**Comment by rickchengx** [comment]
Related Doc will be modified after this PR is reviewed

**Comment by SbloodyS** [comment]
> Related Doc will be modified after this PR is reviewed

Please add the docs in this PR. We suggest that the documents involved in the modification should be submitted together.

**Comment by rickchengx** [comment]
> > Related Doc will be modified after this PR is reviewed
> 
> Please add the docs in this PR. We suggest that the documents involved in the modification should be submitted together.

Sure, but there are many documents that need to be modified, maybe we can first review whether this PR is appropriate

**Comment by SbloodyS** [comment]
> > > Related Doc will be modified after this PR is reviewed
> > 
> > 
> > Please add the docs in this PR. We suggest that the documents involved in the modification should be submitted together.
> 
> Sure, but there are many documents that need to be modified, maybe we can first review whether this PR is appropriate

That sound good to me.

**Comment by fuchanghai** [comment]
If there is only one version of spark on a machine, there is no problem keeping only one. Problems can arise if a machine has two versions. Of course, I don't know about spark, I just analyze this problem from a business perspective

**Comment by fuchanghai** [comment]
@SbloodyS may be this PR is conflict with #11720 ,According to the actual application scenario, it seems that the current PR should be supported. if ds want to support two different versions ,i support another

**Comment by rickchengx** [comment]
Hi, @fuchanghai, Thanks for the reply. I agree with your point. 
1. I think that the current `spark version` on ui is kind of **misleading**. Users may be concerned that DS does not support `spark 3.x.x`. I have seen people asking whether DS supports spark3 , even if there is a description for this problem in the DS [documentation](https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/guide/installation/kubernetes.html) (but users will not find this description easily) as below:

截屏2022-09-09 09 55 13


2. Even if DS needs to support multiple sparks, I don't think setting `SPARK_HOME1` and `SPARK_HOME2` is an elegant way. What if the user needs to support 3 or more sparks?

3. I think that using `SPARK_HOME` is an easy and efficient way without misunderstandings and some potential problems.

**Comment by SbloodyS** [comment]
> @SbloodyS may be this PR is conflict with #11720 ,According to the actual application scenario, it seems that the current PR should be supported. if ds want to support two different versions ,i support another

I think it's better to use current environment management to manage different version of spark. This is the meaning of environmental management.

**Comment by rickchengx** [comment]
> > @SbloodyS may be this PR is conflict with #11720 ,According to the actual application scenario, it seems that the current PR should be supported. if ds want to support two different versions ,i support another
> 
> I think it's better to use current environment management to manage different version of spark. This is the meaning of environmental management.


@SbloodyS Thanks for your suggestion. I totally agree with you.

So DS uses `SPARK_HOME/bin/spark-submit` to build the command, and user can set the `SPARK_HOME` he wants in the environment.

截屏2022-09-09 10 13 33

**Comment by SbloodyS** [comment]
> > > @SbloodyS may be this PR is conflict with #11720 ,According to the actual application scenario, it seems that the current PR should be supported. if ds want to support two different versions ,i support another
> > 
> > 
> > I think it's better to use current environment management to manage different version of spark. This is the meaning of environmental management.
> 
> @SbloodyS Thanks for your suggestion. I totally agree with you.
> 
> So DS uses `SPARK_HOME/bin/spark-submit` to build the command, and user can set the `SPARK_HOME` he wants in the environment.
> 
> 截屏2022-09-09 10 13 33

Yes. That's what I mean.

**Comment by rickchengx** [comment]
> Yes. That's what I mean.

Hi, @SbloodyS , @fuchanghai 

Since #11721 is merged, I've rebased my PR and also modified the related docs.

So now this PR uses `SPARK_HOME/bin/spark-submit` or `SPARK_HOME/bin/spark-sql` to build the command, and user can set the `SPARK_HOME` he wants in the environment as below.

截屏2022-09-13 14 40 34


So the misleading `spark version` is removed, and different tasks can still use different `spark versions` through the environment.

File: docs/docs/en/guide/task/spark.md [hunk 1]
```diff
@@ -15,34 +15,33 @@ Spark task type for executing Spark application. When executing the Spark task,
...
+| Environment Name           | Configure the environment in which to run the script.                                                                                                                                                                         |
+| Number of failed retries   | The number of times the task is resubmitted after failure. It supports drop-down and manual filling.                                                                                                                          |
+| Failure Retry Interval     | The time interval for resubmitting the task if the task fails. It supports drop-down and manual filling.                                                                                                                      |
+| Timeout alarm              | Check Timeout Alarm and Timeout Failure. When the task exceeds the "timeout duration", an alarm email will be sent and the task execution will fail.                                                                          |
```
Comment (zhongjiajie) [inline docs/docs/en/guide/task/spark.md]: It is seems we already change the general parameter into separate parameter files, can you rebase to latest and take a look?

**Review by zhongjiajie — COMMENTED** [review]
please rebase the latest code, our docs make some change
and can you also add some discription in both https://github.com/apache/dolphinscheduler/blob/4dca488cd50b4392d222167c01ae2a79fd295e77/dolphinscheduler-python/pydolphinscheduler/UPDATING.mdand https://github.com/apache/dolphinscheduler/blob/3aa9f2ea25ca42112141aad85140a72b0963e2c3/docs/docs/en/guide/upgrade/incompatible.md becaus this is incompatible change for  users

File: docs/docs/en/guide/task/spark.md (hunk 1 above)
Comment (rickchengx) [reply docs/docs/en/guide/task/spark.md]: Sure.

**Comment by rickchengx** [comment]
> please rebase the latest code, our docs make some change and can you also add some discription in both https://github.com/apache/dolphinscheduler/blob/4dca488cd50b4392d222167c01ae2a79fd295e77/dolphinscheduler-python/pydolphinscheduler/UPDATING.mdand https://github.com/apache/dolphinscheduler/blob/3aa9f2ea25ca42112141aad85140a72b0963e2c3/docs/docs/en/guide/upgrade/incompatible.md becaus this is incompatible change for users

Hi, @zhongjiajie , thanks for your comment. I've rebased my PR and add some descriptions in `UPDATING.md` and `incompatible.md`

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks

**Comment by zhongjiajie** [comment]
@caishunfeng should we release this PR in version 3.1.0?

**Pull request merged**